crawl-ref/source/viewmap.cc

```diff
@@ -1345,6 +1358,7 @@ bool show_map(level_pos &lpos,
     tiles.set_map_display(false);
 #endif
 
+    crawl_state.in_map_mode = false;
```

Comment:
Technically I think you should restore it to what it was previously, not to `false`. A lua function may cause us to enter map mode while already in map mode, somehow. Happily, this is what `unwind_bool` will do anyway, so this won't matter.